test/unit/test_threaded_downloader.py

```diff
@@ -149,6 +149,27 @@ def test_single_download_failure(self):
         self.assertEqual(len(lst.failed_reports), 1)
         self.assertTrue(lst.failed_reports[0].error_msg is not None)
 
+    def test_download_unhandled_exception(self):
+        with mock.patch('nectar.downloaders.threaded._logger') as mock_logger:
+            cfg = config.DownloaderConfig()
+            lst = listener.AggregatingEventListener()
+            downloader = threaded.HTTPThreadedDownloader(cfg, lst)
+
+            URL = 'http://pulpproject.org/robots.txt'
```

Comment:
I don't think our tests should make actual requests. Having this external URL makes nectar tests depend on multiple external things to be true (DNS, internet access, pulproject.org, robots.txt on that site, http vs https). Please use mocks to ensure this test runs but doesn't actually make and requests using `python-requests`. Probably mocking `python-requests` objects is the best way to do this.